multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v3/ProvidedDependencyReferenceResolver.java

```diff
@@ -0,0 +1,60 @@
+package org.cloudfoundry.multiapps.mta.resolvers.v3;
+
+import static org.cloudfoundry.multiapps.mta.resolvers.ReferencePattern.FULLY_QUALIFIED;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.cloudfoundry.multiapps.common.ContentException;
+import org.cloudfoundry.multiapps.mta.handlers.v2.DescriptorHandler;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.NamedElement;
+import org.cloudfoundry.multiapps.mta.model.ProvidedDependency;
+import org.cloudfoundry.multiapps.mta.resolvers.ProvidedValuesResolver;
+import org.cloudfoundry.multiapps.mta.resolvers.ProvidesValuesResolver;
+import org.cloudfoundry.multiapps.mta.resolvers.ReferenceResolver;
+import org.cloudfoundry.multiapps.mta.resolvers.ResolverBuilder;
+
+public class ProvidedDependencyReferenceResolver extends ReferenceResolver<ProvidedDependency> {
+
+    protected final ProvidedDependency dependency;
+    protected final ResolverBuilder propertiesResolverBuilder;
+    private DescriptorHandler handler;
+    private DeploymentDescriptor descriptor;
+    private String consumerName;
+
+    public ProvidedDependencyReferenceResolver(DeploymentDescriptor descriptor, NamedElement container, ProvidedDependency dependency,
+                                               String prefix, ResolverBuilder propertiesResolverBuilder,
+                                               Set<String> dynamicResolvableParameters) {
+        this(descriptor, container, dependency, prefix, new DescriptorHandler(), propertiesResolverBuilder, dynamicResolvableParameters);
+    }
+
+    public ProvidedDependencyReferenceResolver(DeploymentDescriptor descriptor, NamedElement container, ProvidedDependency dependency,
+                                               String prefix, DescriptorHandler handler, ResolverBuilder propertiesResolverBuilder,
+                                               Set<String> dynamicResolvableParameters) {
+        super(dependency.getName(), prefix, handler, descriptor, container.getName(), FULLY_QUALIFIED, dynamicResolvableParameters);
+        this.dependency = dependency;
+        this.propertiesResolverBuilder = propertiesResolverBuilder;
+        this.handler = handler;
+        this.consumerName = container.getName();
```

Comment:
`container` is a bit hard to understand here. Maybe `parent`? Or sticking with the class field name, maybe `consumer`?